assembly/src/parsers/io_ops/adv_ops.rs

```diff
@@ -53,6 +53,25 @@ pub fn parse_adv_loadw(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(),
     Ok(())
 }
 
+/// Appends operations to the span block to execute `adv_pipe` instruction. The sequence of
+/// operations is: PIPE RPPERM.
+///
+/// This instruction requires 2 VM cycles to execute.
+///
+/// # Errors
+/// This function will return an `AssemblyError` if the `adv_pipe` instruction is malformed.
+pub fn parse_adv_pipe(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
```

Comment:
We should probably also add a parser for this in the new AST structure (e.g., [here](https://github.com/maticnetwork/miden/blob/next/assembly/src/parsers/ast/context.rs#L452)) and a new `Instruction` node [here](https://github.com/maticnetwork/miden/blob/next/assembly/src/parsers/ast/nodes.rs#L233).

For `mem_stream`, I actually added an `Instruction` node but forgot to add a parser - could you add one for `mem_stream` too?